### PR TITLE
Handle release through git tag

### PR DESCRIPTION
**Pull request by crazy-max**
| Event           | Ref                           | Commit SHA | Docker Tags                         | Push |
|-----------------|-------------------------------|------------|-------------------------------------|--------|
| `pull_request`  | `refs/pull/2/merge`           | `a123b57`  | `pr-2`, `buildkit-pr-2`             | :x: |
| `push`          | `refs/heads/master`           | `cf20257`  | `master`, `buildkit-master`         | :white_check_mark: |
| `push`          | `refs/heads/my/branch`        | `a5df687`  | `my-branch`, `buildkit-my-branch`   | :white_check_mark: |
| `push tag`      | `refs/tags/deploy/1`          | `ad132f5`  | `1`, `latest`, `buildkit-1`, `buildkit-latest`       | :white_check_mark: |

Basically this PR allows to release the images without a deployment (deploy.yml). `latest` is generated only if a git tag event is triggered and starts with `deploy/*`. Would have like to use [`regex`](https://www.terraform.io/docs/configuration/functions/regex.html) with [`can`](https://www.terraform.io/docs/configuration/functions/can.html) but `can` doesn't seem available in the current stdlib (`can(regex("^v\\d+", tag))`):

```
error: docker-bake.hcl:45,64-73: Error in function call; Call to function "gen-tags" failed: docker-bake.hcl:16,13-16: Call to unknown function; There is no function named "can".., and 1 other diagnostic(s)
```

**EDIT:** Will be possible to use `can` through docker/buildx#491

### CODE REVIEW

**Comment by crazy-max** [comment]
@tonistiigi PR updated following what we said. If you want we can go full bake hack scripts. Let me know if you're interested.
